clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -3169,6 +3169,50 @@ check_kernel_syscall_trace(void)
 #endif
 }
 
+bool
+check_has_instructions(void)
+{
+    std::cerr << "Testing at-least-1-instruction\n";
+    // Correct: 1 regular instruction.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_A),
+            gen_exit(TID_A),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Correct: 1 unfetched instruction.
```

Comment:
(Optional) Some coverage analyses would like to see a test for TRACE_TYPE_PREFETCH_INSTR too.

Reply:
Hmm somehow this didn't show up in the page and it said no conversations and it only appeared after merging...strange, but not just a client cache issue b/c the server shouldn't let it merge w/ unresolved conversations.  I think for this we will live with just testing the other type.